zebra-network/src/protocol/external/codec.rs

```diff
@@ -35,12 +35,14 @@ use super::{
 const HEADER_LEN: usize = 24usize;
 
 /// A codec which produces Bitcoin messages from byte streams and vice versa.
+#[derive(Clone)]
```

Comment:
These `Clone` derivations were added so that the `new_test_connection` function can clone the `Codec` in order to create separate `FrameRead` and `FrameWrite` types. Both of them require a `Codec` instance, one to decode `Message`s and the other to write `Message`s.

Reply:
I'm resolving this because I managed to remove it. I replaced the `Framed` channels with a single `mpsc` unbounded channel, which should also make the tests a little faster because it avoids serialization and deserialization of messages.